switchbot/devices/curtain.py

```diff
@@ -121,10 +147,21 @@ async def get_basic_info(self) -> dict[str, Any] | None:
             "solarPanel": bool(_data[5] & 0b00001000),
             "calibration": bool(_data[5] & 0b00000100),
             "calibrated": bool(_data[5] & 0b00000100),
-            "inMotion": bool(_data[5] & 0b01000011),
-            "position": (100 - _position) if self._reverse else _position,
+            "inMotion": _in_motion,
+            "position": _direction_adjusted_position,
             "timers": _data[7],
         }
+    
+    def _update_motion_direction(self, in_motion: bool, previous_position: int|None, new_position: int) -> None:
+        """Update opening/closing status based on movement."""
+        if previous_position is None:
+            return
+        if in_motion == True:
```

Comment:
```suggestion
        if in_motion is True:
```

Use identity checks for singletons